### PR TITLE
Only use buffer on rename and show usages

### PR DESCRIPTION
**Pull request by ethan-leba**
Here's my attempt on #1017. It's mostly working, the only issue I ran into is that the error `E21: Cannot make changes, 'modifiable' is off` always pops up before the quickfix list opens to show usages after refactoring. Other than that, it works, but I'm pretty stumped on this bug so I thought I'd just open the PR and see if any of the maintainers know what's going on here!

### CODE REVIEW

**Comment by ethan-leba** [comment]
Looks like master is failing CI

**Comment by davidhalter** [comment]
Master passed: https://travis-ci.org/github/davidhalter/jedi-vim/builds/659627955. Not sure if something on the CI changed.

However, I'm still pretty sure that this PR doesn't have anything to do with the failure, so we should probably just change this test (it has been annoying before).

**Comment by ethan-leba** [comment]
Sorry, worded that poorly. What I meant by that was that running tests on master locally failed for me

File: pythonx/jedi_vim.py [hunk 1]
```diff
@@ -1027,11 +1030,15 @@ def do_rename(replace, orig=None):
...
 
+    vim_command('w')
+    vim.current.buffer = saved_buffer
+    vim.current.window.cursor = saved_cursor
     # Restore previous tab and window.
     vim_command('tabnext {0:d}'.format(saved_tab))
     vim_command('{0:d}wincmd w'.format(saved_win))
 
     if len(buffers) > 1:
+        usages()
```
Comment (davidhalter) [inline pythonx/jedi_vim.py:1041]: Why is this necessary?

File: pythonx/jedi_vim.py [hunk 2]
```diff
@@ -1074,11 +1081,11 @@ def py_import_completions():
 
 
 @catch_and_print_exceptions
-def new_buffer(path, options='', using_tagstack=False):
+def new_buffer(path, options='', using_tagstack=False, force_buffer=False):
     # options are what you can to edit the edit options
-    if int(vim_eval('g:jedi#use_tabs_not_buffers')) == 1:
+    if int(vim_eval('g:jedi#use_tabs_not_buffers')) == 1 and not force_buffer:
```
Comment (davidhalter) [inline pythonx/jedi_vim.py:1086]: For tabs the old behavior is fine IMO. I use it regularly.

File: pythonx/jedi_vim.py [hunk 3]
```diff
@@ -1027,11 +1030,15 @@ def do_rename(replace, orig=None):
         vim.current.buffer[r.line - 1] = (r_line[:r.column] + replace +
                                           r_line[r.column + len(orig):])
 
+    vim_command('w')
```
Comment (davidhalter) [inline pythonx/jedi_vim.py:1033]: I don't think we should do this. We should not save randomly. This is the user's responsibility.

File: pythonx/jedi_vim.py [hunk 4]
```diff
@@ -444,6 +444,7 @@ def show_goto_multi_results(definitions, mode):
             # well.
             lst.append(dict(text=PythonToVimStr(d.description)))
         else:
+            # EL: Error here
```
Comment (davidhalter) [inline pythonx/jedi_vim.py:447]: What does this mean?

**Comment by davidhalter** [comment]
CI should be working smooth now.

File: pythonx/jedi_vim.py (hunk 2 above)
Comment (ethan-leba) [reply pythonx/jedi_vim.py:1086]: You're right, will fix this :+1:

File: pythonx/jedi_vim.py (hunk 1 above)
Comment (ethan-leba) [reply pythonx/jedi_vim.py:1041]: Was following blueyed's suggestion here: https://github.com/davidhalter/jedi-vim/issues/1017#issuecomment-647227001

File: pythonx/jedi_vim.py (hunk 3 above)
Comment (ethan-leba) [reply pythonx/jedi_vim.py:1033]: That makes sense, but for creating new buffers the function `new_buffer` saves anyways for all buffers except the final one, and the buffer needs to be saved in order to return to the original buffer where the refactor happened.

Comment (ethan-leba) [reply pythonx/jedi_vim.py:1033]: I should disable this behavior for tabs though

File: pythonx/jedi_vim.py (hunk 4 above)
Comment (ethan-leba) [reply pythonx/jedi_vim.py:447]: That was where I thought the bug I was referring to in the description above was coming from, but I narrowed it down to this: https://github.com/davidhalter/jedi-vim/blob/master/autoload/jedi.vim#L484

Comment (ethan-leba) [reply pythonx/jedi_vim.py:447]: Still have no clue why an error is being emitted 😬

File: pythonx/jedi_vim.py (hunk 3 above)
Comment (blueyed) [reply pythonx/jedi_vim.py:1033]: Saving should at least get skipped with `set hidden` then maybe (a sane default).
